smart-freight-center/sfc-backend
# Chore adds unit tests for the Share footprint Uscase

## Description

During the PoC, we completely ignored writing unit and integration tests because we were not sure of the features and there was not clear path on how it would work. 

Having improved the usecses to use interfaces in #138  and #139, it is not possible(and easier) to add unit tests for our usecases. 

This PR does exactly that. It add tests for the share-footprint usecase. It can be used as a guide on how we create mock instances of the interfaces(using sinon) and simulate the different possible behaviours in our tests.

### How to test it
If you run 

```bash
npm run test:unit
```

All unit tests should run. 

Note that 

```bash
npm run tdd:unit
```
Would run the tests anytime you make a change to the file

## Approach
- To make it easier, the first thing was to ensure that we are using interfaces in all the dependencies of our usecases. In this PR we created interfaces for the SfcAPI
- Using [sinon](https://www.npmjs.com/package/sinon) we created stubs for these interfaces(see ___tests__/stubs.ts)
- Then added tests for the input validation and data model verification taking advantage of the stubs we created

Resolves #134
